aten/src/ATen/native/quantized/cpu/qadd.cpp

```diff
@@ -9,48 +9,62 @@
+}
+
+// Note: out is assumed to be the same size as self and other.
 template <bool ReLUFused = false>
-class QAddInt8 final : public c10::OperatorKernel {
- public:
-  Tensor operator()(
-      at::Tensor qa,
-      at::Tensor qb,
-      double scale,
-      int64_t zero_point) {
-    TORCH_CHECK(
-        qa.numel() == qb.numel(), "Add operands must be the same size!");
-    TORCH_CHECK(
-        qa.scalar_type() == qb.scalar_type(),
-        "Add operands should have same data type.");
-    auto a = qa.dequantize();
-    auto b = qb.dequantize();
-    auto c = at::empty_like(a);
-    auto iter = TensorIterator::binary_op(c, a, b);
+Tensor _add_out(Tensor& out, const Tensor& self, const Tensor& other) {
+  long zero_point = out.q_zero_point();
+  double scale = out.q_scale();
+  // Requantize a and b into int32.
+  auto iter = TensorIterator::binary_op(out, self, other);
+  AT_DISPATCH_QINT_TYPES(out.scalar_type(), "qadd", [&]() {
+    cpu_kernel(iter, [&](scalar_t a, scalar_t b) -> scalar_t {
+      const auto da = at::dequantize_val(self.q_scale(), self.q_zero_point(), a);
+      const auto db = at::dequantize_val(other.q_scale(), other.q_zero_point(), b);
+      float c = da + db;
+      if (ReLUFused) {
+        c = std::max<float>(c, 0.0);
+      }
+      return at::quantize_val<scalar_t>(scale, zero_point, c);
+    });
+  });
+  return out;
+}
 
-    if (ReLUFused) {
-      cpu_kernel(iter, [&](float a_val, float b_val) -> float {
-        return std::max<float>(a_val + b_val, 0);
-      });
-    } else {
-      cpu_kernel(iter, [&](float a_val, float b_val) -> float {
-        return a_val + b_val;
-      });
-    }
-    return at::quantize_linear(
-        c, scale, zero_point, qa.scalar_type()); // Requantize
+template <bool ReLUFused = false>
+class QAdd final : public c10::OperatorKernel {
+ public:
+  Tensor operator()(at::Tensor qa, at::Tensor qb,
+                    double scale, int64_t zero_point) {
+    TORCH_CHECK(qa.numel() == qb.numel(),
+                "Add operands must be the same size!");
+    TORCH_CHECK(qa.scalar_type() == qb.scalar_type(),
+                "Add operands should have same data type.");
+    auto qc = at::_empty_affine_quantized(qa.sizes(),
+      at::device(kCPU).dtype(qa.scalar_type()),
+      scale, zero_point);
+    return _add_out<ReLUFused>(qc, qa, qb);
   }
 };
 
-static auto registry =
-    c10::RegisterOperators()
-        .op("quantized::add(Tensor qa, Tensor qb, float scale, int zero_point)"
-            "-> Tensor qc",
-            c10::RegisterOperators::options()
-                .kernel<QAddInt8</*ReLUFused=*/false>>(QuantizedCPUTensorId()))
-        .op("quantized::add_relu(Tensor qa, Tensor qb, float scale, int zero_point)"
-            "-> Tensor qc",
-            c10::RegisterOperators::options()
-                .kernel<QAddInt8</*ReLUFused=*/true>>(QuantizedCPUTensorId()));
-} // namespace
-} // namespace native
-} // namespace at
+static auto registry = c10::RegisterOperators()
```

Comment:
Did you change anything here?

Reply:
Yes, minor refactor in the formatting to allow better usage of limited line-length